### ISSUE TITLE
Issues with handling of virtual columns

### ISSUE THREAD

**Issue by richardeaxon** [HeidiSQL/HeidiSQL#1210]
**Describe the bug**

Having a number of issues dealing with virtual columns in MariaDB 10.4

**To Reproduce**

Using the following table schema:

```
DROP table IF EXISTS ip_addresses;

CREATE TABLE `ip_addresses` (
`network` CHAR(50) NULL DEFAULT 'default' COMMENT 'Network this address belongs to' COLLATE 'utf8_general_ci',
`inet_n` VARBINARY(16) NOT NULL COMMENT 'Subnet address',
`prefix` TINYINT(3) UNSIGNED NOT NULL COMMENT 'Prefix',
    `inet_a` VARCHAR(39) AS (inet6_ntoa(`inet_n`)) VIRTUAL,
    `port_name` VARCHAR(100) AS (CONCAT(`network`, ' ', INET6_NTOA(`inet_n`), '/', `prefix`)) VIRTUAL,

UNIQUE INDEX `network` (`network`, `inet_n`, `prefix`) USING BTREE
)
COMMENT='NOTE: collation for VARBINARY field MUST be utf8mb4_unicode_520_ci',
COLLATE='utf8mb4_unicode_520_ci'
ENGINE=InnoDB;
```

And then using HeidiSQL to show table CREATE code results in the following:

```
CREATE TABLE `ip_addresses` (
	`network` CHAR(50) NULL DEFAULT 'default' COMMENT 'Network this address belongs to' COLLATE 'utf8_general_ci',
	`inet_n` VARBINARY(16) NOT NULL COMMENT 'Subnet address',
	`prefix` TINYINT(3) UNSIGNED NOT NULL COMMENT 'Prefix',
	`inet_a` VARCHAR(39) DEFAULT NULL AS (inet6_ntoa(`inet_n`)) virtual COLLATE 'utf8mb4_unicode_520_ci',
	`port_name` VARCHAR(100) DEFAULT NULL AS (concat(`network`,' ',inet6_ntoa(`inet_n`),'/',`prefix`)) virtual COLLATE 'utf8mb4_unicode_520_ci',
	UNIQUE INDEX `network` (`network`, `inet_n`, `prefix`) USING BTREE
)
COMMENT='NOTE: collation for VARBINARY field MUST be utf8mb4_unicode_520_ci'
COLLATE='utf8mb4_unicode_520_ci'
ENGINE=InnoDB
;
```

This is an invalid table schema as virtual columns do not have attributes like unsigned, null, collate etc. Trying to recreate the schema results in an error. The same issues occur if you attempt to alter a virtual column.

**Expected behavior**
VIRTUAL columns for CREATE and ALTER should not have any additional attributes other than the column definition and expression. VIRTUAL columns do support COMMENT.

**Screenshots**

**Environment:**
 - OS: [Centos 7.4]
 - HeidiSQL version [e.g. 11.0.0.6099]
 - Database type and version [e.g. MariaDB 10.4.14]

**Comment by ansgarbecker** [comment]
I suppose the collation needs to be silenced for virtual columns. Probably also for all columns which have the same collation as the table itself, as they anyway derive it from the table.

**Comment by Webifi** [comment]
Looks like this issue keeps getting punted.  Any actual plans to fix it or will it continue to be punted?

Moreover, this same (or closely related) issue makes it impossible to alter VIRTUAL columns in HeidiSQL.

**Comment by ansgarbecker** [comment]
`DEFAULT x` and `COLLATE x` clauses removed for virtual columns in next build.
Also fixes #1497
